src/sentry/models/openidtoken.py

```diff
@@ -0,0 +1,47 @@
+from datetime import timedelta
+
+from django.db import models
+from django.utils import timezone
+
+from sentry.db.models import FlexibleForeignKey, Model, control_silo_only_model
+from sentry.utils import jwt as jwt_utils
+
+DEFAULT_EXPIRATION = timedelta(minutes=10)
+
+
+def default_expiration():
+    return timezone.now() + DEFAULT_EXPIRATION
+
+
+@control_silo_only_model
+class OpenIDToken(Model):
+    """
+    ID Token for a specific user issued as a result of user authentication.
+    Compliant with the OpenID Connect Core 1.0 spec
+    """
+
+    __include_in_export__ = False
+
+    user = FlexibleForeignKey("sentry.User")
+
+    iss = models.CharField(max_length=64, default="https://sentry.io")
+    aud = models.CharField(max_length=64)
+    exp = models.DateTimeField(db_index=True, default=default_expiration)
+    iat = models.DateTimeField(default=timezone.now)
+    nonce = models.CharField(max_length=64, null=True)
```

Comment:
Good point actually - I'd still like to give them their own class in our backend (so we can reuse them wherever), so I'll modify this such that it's just a regular class and not linked to our DB

Reply:
It is true that JWT claims need not be stored in the database.

If in the future you plan to have a mechanism to be able to invalidate JWT values, then storing them in the database could be useful.

Reply:
Generally OIDC ID tokens represent one successful authentication by the user at a certain time - it's not used to access API's or anything, it also generally holds infrequently-changing information that the client will store on their end, and then later they might fetch another token if they want to see what's changed. Because of this I think storing them isn't necessary now, and if we have any reason to change that later it won't be very difficult.

Reply:
Moved it into a standard Python class - thoughts on this approach? @markstory @dashed